architecture/build-service.md

```diff
@@ -19,170 +19,114 @@ The Build Service is dependent on the following services:
 
 The Build Service contains these controllers:
 - Component Build Controller
-  - Monitors Component CRs and creates PipelineRuns - directly or via [Pipelines As Code (PaC)](https://pipelinesascode.com) provided by Pipeline Service.
+  - Monitors Component CRs and creates PipelineRun definitions which will be used by [Pipelines As Code (PaC)](https://pipelinesascode.com) provided by Pipeline Service.
 - PaC PipelineRun Pruner Controller
-  - Deletes PipelineRuns managed by Pipelines As Code for Components that were deleted.
-- Git Tekton Resources Renovater
-  - Provides updates for the .tekton directory in the user repository which has been created by Component Build Controller in Custom Mode.
+  - Deletes PipelineRuns for Components that were deleted.
+- Component dependency update controller
+  - Monitors push PipelineRuns and based on set relationships runs renovate which updates
+    SHA for image from PipelineRun in user's repository.
 
 ### Component Build Controller
 
-Component Build Controller is managed by Component CR changes. It's using Component CR annotations and `.status.devfile` for selecting a mode and configuration of the PipelineRuns.
+Component Build Controller is managed by Component CR changes (creation or update).
+It's using Component CR annotations and configuration of the PipelineRuns.
 
 #### Modes
+The prerequisite is to have installed GitHub App which is used by the Build Service in the user's repository, or have gitlab/github secret created for usage via webhook.
```

Comment:
Does it make sense to point to Konflux doc on how to create secrets?

Reply:
added